run-tests.py

```diff
@@ -148,6 +152,163 @@ def doctest_examples_readme():
         print('All example notebooks are indexed.')
 
 
+def doctest_rst_and_public_interface():
+    """
+    Check that every class and method is documented in an rst file and that
+    no unintended modules are exposed via a public interface
+    """
+    print('\nChecking that all classes and methods are documented in an RST '
+          'file and that public interfaces are clean.')
+
+    # Import all public pints modules. Importing everything here is not
+    # strictly needed: `import pints` happens to import `pints.noise`, and
+    # `import pints.toy` will import `pints` and `pints.toy`, but we list
+    # everything here for completeness. If a new module is added to pints
+    # it should be imported here for this doctest.
+    import pints
+    import pints.io
+    import pints.noise
+    import pints.plot
+    import pints.toy
+
+    # If any modules other than these are exposed it may indicate that a module
+    # has been inadvertently exposed in a public context, or that a new module
+    # has been added to pints and should be imported above and included in this
+    # list.
+    pints_submodules = ['pints.io', 'pints.noise', 'pints.plot', 'pints.toy']
+
+    doc_symbols = get_all_documented_symbols()
+
+    check_exposed_symbols(pints, pints_submodules, doc_symbols)
+    check_exposed_symbols(pints.io, [], doc_symbols)
+    check_exposed_symbols(pints.noise, [], doc_symbols)
+    check_exposed_symbols(pints.plot, [], doc_symbols)
+    check_exposed_symbols(pints.toy, [], doc_symbols)
+
+    print('All classes and methods are documented in an RST file, and all '
+          'public interfaces are clean.')
+
+
+def check_exposed_symbols(_module, _submodule_names, _doc_symbols):
+    """
+    Check _module for any classes and methods not contained in _doc_symbols,
+    and check for any modules not contained in _submodule_names.
+
+    :param _module: the module to check
```

Comment:
This is cool, but we're not using it at the moment and not sure if our template supports it. If we start using this it'll have to be a pints-wide change in a separate commit. For now please do:

```
Arguments:

``module``
    The module to check
``submodule_names``
    List of submodules exposed by ``module``
```
etc.

(that creates a definition list)